dxr/plugins/core.py

```diff
@@ -306,6 +311,8 @@ def needles(self):
         extension = splitext(self.path)[1]
         if extension:
             yield 'ext', extension[1:]  # skip the period
+        if is_image(self.path):
```

Comment:
Doing a contains_text() check will filter out .svg images since they are just XML files (right now they are previewed just like binary image types).
Currently the "is it an image" check is always done through is_image, which queries the extension.